app/src/androidTest/java/com/example/theperiodpurse/NavigationTest.kt

```diff
@@ -0,0 +1,70 @@
+package com.example.theperiodpurse
+
+import androidx.activity.ComponentActivity
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onNodeWithContentDescription
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.navigation.compose.ComposeNavigator
+import androidx.navigation.testing.TestNavHostController
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class NavigationTest {
+    @get:Rule
+    val composeTestRule = createAndroidComposeRule<ComponentActivity>()
+    private lateinit var navController: TestNavHostController
+
+    @Before
+    fun setupNavHost() {
+        composeTestRule.setContent {
+            navController =
+                TestNavHostController(LocalContext.current)
+            navController.navigatorProvider.addNavigator(
+                ComposeNavigator()
+            )
+            ScreenApp(navController = navController)
+        }
+    }
+
+    @Test
+    fun appNavhost_clickSettings_navigatesToSettingsScreen() {
+        composeTestRule.onNodeWithText(Screen.Settings.name)
+            .performClick()
+        navController.assertCurrentRouteName(Screen.Settings.name)
+    }
+
+    private fun navigateToSettingsScreen() {
+        composeTestRule.onNodeWithText(Screen.Settings.name).performClick()
+    }
+
+    @Test
+    fun appNavhost_clickSettings_navigatesToInfoScreen() {
+        composeTestRule.onNodeWithText(Screen.Learn.name)
+            .performClick()
+        navController.assertCurrentRouteName(Screen.Learn.name)
+    }
+
+    private fun navigateToInfoScreen() {
+        composeTestRule.onNodeWithText(Screen.Learn.name).performClick()
+    }
+
+    @Test
+    fun appNavHost_clickCalendarFABOnSettingsScreen_navigatesToCalendarScreen() {
+        navigateToSettingsScreen()
+        composeTestRule.onNodeWithContentDescription("fab")
```

Comment:
There might be a better way of getting the FAB node. It doesn't have text on it so I had to use something like contentDescription. We could look into test tags/making contentDescriptions more standardized